support-frontend/app/services/stepfunctions/SupportWorkersClient.scala

```diff
@@ -154,7 +154,7 @@ class SupportWorkersClient(
         user = user,
         giftRecipient = giftRecipient,
         product = request.body.product,
-        paymentProvider = PaymentProvider.fromPaymentFields(request.body.paymentFields.left.toOption),
+        analyticsInfo = AnalyticsInfo(giftRecipient.isDefined, PaymentProvider.fromPaymentFields(request.body.paymentFields.left.toOption)),
```

Comment:
* add new isGiftPurchase field so that sendacquisition/failure lambda can still recognise gifts of any type
* wrap paymentProvider and isGiftPurchase fields that are only used by acquisition/failure lambda in dedicated AnalyticsInfo object for clarity and ease of use

I had to keep it all on one line as this method is already at the 50 line limit 😢